analyzers/src/SonarAnalyzer.Common/Rules/StringLiteralShouldNotBeDuplicatedBase.cs

```diff
@@ -46,7 +46,7 @@ public abstract class StringLiteralShouldNotBeDuplicatedBase<TSyntaxKind, TLiter
         protected abstract bool IsMatchingMethodParameterName(TLiteralExpressionSyntax literalExpression);
         protected abstract bool IsInnerInstance(SyntaxNodeAnalysisContext context);
         protected abstract IEnumerable<TLiteralExpressionSyntax> RetrieveLiteralExpressions(SyntaxNode node);
-        protected abstract string GetLiteralValue(TLiteralExpressionSyntax literal);
+        protected abstract SyntaxToken GetLiteralToken(TLiteralExpressionSyntax literal);
```

Comment:
I don't like the `Get` prefixes. Every function returns (gets) something.
```suggestion
        protected abstract SyntaxToken LiteralToken(TLiteralExpressionSyntax literal);
```

Reply:
Speaking of that, you can rename `RetrieveLiteralExpressions` to `LiteralExpressions` :D Or `FindLiteralExpressions` to express that it is more costly with some logic inside